core/src/main/java/io/questdb/griffin/SqlCompiler.java

```diff
@@ -1441,7 +1441,9 @@ private CompiledQuery repairTables(SqlExecutionContext executionContext) throws
             tableExistsOrFail(lexer.lastTokenPosition(), tok, executionContext);
 
             try {
-                engine.getWriter(executionContext.getCairoSecurityContext(), tok).close();
+                TableWriter writer = engine.getWriter(executionContext.getCairoSecurityContext(), tok);
```

Comment:
i'm wondering if it would be better to make it responsibility of the reader to lock on writer? In this case migrating it a simpler engine call.

Additionally writer must also be closed in case of exception